client/my-sites/email/email-providers-stacked-comparison/provider-cards/price-with-interval.tsx

```diff
@@ -13,44 +14,72 @@ type PriceWithIntervalProps = {
 };
 
 const PriceWithInterval: FunctionComponent< PriceWithIntervalProps > = ( props ) => {
-	const { className, cost, currencyCode, hasDiscount, sale, intervalLength } = props;
-
-	const priceClassName = `${ className }__${
-		hasDiscount ? 'discounted-price' : 'keep-main-price'
-	}`;
-	const intervalLengthClassName = `${ className }__interval`;
-	const mailboxClassName = `${ className }__mailbox`;
-	const saleClassName = `${ className }__sale`;
+	const { cost, currencyCode, hasDiscount, sale, intervalLength } = props;
 
 	const showSale = sale && sale !== 0;
 
-	const lineThrough = {
-		textDecoration: hasDiscount || showSale ? 'line-through' : 'solid ',
+	const priceClassName = classNames( 'provider-cards__price-with-interval-price', {
+		'provider-cards__price-with-interval-price-discounted': hasDiscount || showSale,
+	} );
+
+	const priceSpan = <span className={ priceClassName } />;
+	const standardPrice = formatCurrency( cost ?? 0, currencyCode );
+
+	const getSalePriceContents = (): TranslateResult => {
+		const saleTranslateArguments = {
+			salePrice: formatCurrency( sale ?? 0, currencyCode ),
+			standardPrice,
+		};
+		const saleTranslateComponents = {
+			priceSpan,
+			saleSpan: <span className="provider-cards__price-with-interval-sale-price" />,
+		};
+
+		if ( intervalLength === IntervalLength.ANNUALLY ) {
```

Comment:
Could we eliminate this `If` block if we passed the interval ( i.e. month/year ) as a translate argument? If this will not make the translation less accurate,  of course.

Reply:
I think we want to keep the `if` so we can keep more context in the translated strings. We're likely to end up with poorer translations if we translate `yearly/monthly` independently.